ugvc/cnv/CNV_eval.py

```diff
@@ -0,0 +1,162 @@
+import sys
+import argparse
+import pandas as pd
+import os
+import warnings
+warnings.filterwarnings('ignore')
+
+def intersection(sample1_basename,sample2_basename,sample1_bed_file,sample2_bed_file,intersection_cutoff=0.5):
+    out_sample1_specific = f"{sample1_basename}.specific_cnv.bed"
+    out_sample2_specific = f"{sample2_basename}.specific_cnv.bed"
+    out_common_sample1 = f"{sample1_basename}.common_cnv.bed"
+    out_common_sample2 = f"{sample2_basename}.common_cnv.bed"
+
+    cmd = f"bedtools subtract -N -f {str(intersection_cutoff)} -a {sample1_bed_file} -b {sample2_bed_file} > {out_sample1_specific}"
+    os.system(cmd)
+    cmd = f"bedtools subtract -N -f {str(intersection_cutoff)} -a {sample2_bed_file} -b {sample1_bed_file} > {out_sample2_specific}"
+    os.system(cmd)
+    cmd = f"bedtools subtract -A -a {sample1_bed_file} -b {out_sample1_specific} > {out_common_sample1}"
+    os.system(cmd)
+    cmd = f"bedtools subtract -A -a {sample2_bed_file} -b {out_sample2_specific} > {out_common_sample2}"
+    os.system(cmd)
+    
+    count_sample1_specific=0
+    count_sample2_specific = 0
+    count_sample_1_common = 0
+    count_sample_2_common = 0
+    
+    if os.stat(out_sample1_specific).st_size > 0:
+        count_sample1_specific=sum(1 for line in open(out_sample1_specific))
+    if os.stat(out_sample2_specific).st_size > 0:
+        count_sample2_specific = sum(1 for line in open(out_sample2_specific))
+    if os.stat(out_common_sample1).st_size > 0:
+        count_sample_1_common = sum(1 for line in open(out_common_sample1))
+    if os.stat(out_common_sample2).st_size > 0:
+        count_sample_2_common = sum(1 for line in open(out_common_sample2))
+
+    return [count_sample1_specific,count_sample2_specific,count_sample_1_common,count_sample_2_common]
+
+def filter_len(sample1_basename,sample2_basename,len_cutoff=10000):
+    sample1_specific = f"{sample1_basename}.specific_cnv.bed"
+    sample2_specific = f"{sample2_basename}.specific_cnv.bed"
+    common_sample1 = f"{sample1_basename}.common_cnv.bed"
+    common_sample2 = f"{sample2_basename}.common_cnv.bed"
+
+    out_sample1_specific_len = f"{sample1_basename}.specific_cnv.len.bed"
+    out_sample2_specific_len = f"{sample2_basename}.specific_cnv.len.bed"
+    out_sample1_common_len = f"{sample1_basename}.common_cnv.len.bed"
+    out_sample2_common_len = f"{sample2_basename}.common_cnv.len.bed"
+
+    # filter by length
+    cmd = 'awk \'$3-$2>=10000\' ' + sample1_specific + ' > ' + out_sample1_specific_len
```

Comment:
I am normally not a huge pedant about using bash, but on small files like this would be nice to use python